tests/TestHarness/Node.py

```diff
@@ -489,6 +490,34 @@ def waitForTransInBlock(self, transId, timeout=None):
         ret=Utils.waitForBool(lam, timeout)
         return ret
 
+    def checkBlockForTransactions(self, transIds, blockNum):
+        block = self.processCurlCmd("trace_api", "get_block", f'{{"block_num":{blockNum}}}', silentErrors=False, exitOnError=True)
+        if block['transactions']:
+            for trx in block['transactions']:
+                if trx['id'] in transIds:
+                    transIds.pop(trx['id'])
+        return transIds
+
+    def waitOnBlockTransactions(self, transIds, startBlock=2, timeout=30):
```

Comment:
Could you rename `timeout` to be something that suggests it is a ceiling on the number of additional blocks the function is willing to wait to be produced to check for transactions before returning (`"ERROR: Transactions were missing upon expiration of waitOnblockTransactions"`)?